### ISSUE TITLE
Invalid `srcset` being generated by `gatsby-transformer-remark` when using `react-rehype` when processing markdown with images

### ISSUE THREAD

**Issue by Ameobea** [gatsbyjs/gatsby#34338]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

There is a bug somewhere in the complex chain when processing markdown that contains images with `gatsby-transformer-remark` and friends.  This happens when adding and setting up `rehype-react` following the [official Gatsby guide](https://using-remark.gatsbyjs.org/custom-components/) on using custom components.

The generated AST from `gatsby-transformer-remark` and other plugins is generated with `srcSet` props that are arrays of strings.  When this AST is passed into `rehype-react`, the array is not properly turned into a valid `srcset` string - it is missing commas.  This results in the srcset not being processed by my browser (Chrome) and the fallback image src being loaded instead.  I do not know if the bug is in a downstream library (`rehype-react`, `hyperscript`, or something else) or if the `srcSet` array being generated by Gatsby is actually correct.

The reproduction link below is simply the `gatsby-starter-blog` with a very tiny commit added on top to enable `rehype-react`, following the official guide.

### Reproduction Link

https://github.com/Ameobea/gatsby-srcset-bug-repro

### Steps to Reproduce

1. `npx gatsby new gatsby-starter-blog https://github.com/gatsbyjs/gatsby-starter-blog`
2. Apply commit from the repo repository (https://github.com/Ameobea/gatsby-srcset-bug-repro/commit/834e2d9ebb8c4e3ca8fd7e0e528cf808a4678d47)
3. `yarn build` or `yarn start` at http://localhost:8000/hello-world/ and view generated HTML on the "Hello World" blog post.  Note the lack of commas in the `srcset` for the image and errors in the console:

![](https://ameo.link/u/9l9.png)

### Expected Result

Expect that the `srcset` attribute is valid with commas between the items rather than just spaces.  Should look like this:

`<img class="gatsby-resp-image-image" alt="Chinese Salty Egg" title="" src="/static/8058f3f26913fea3b6a89a73344fe94a/828fb/salty_egg.jpg" srcset="/static/8058f3f26913fea3b6a89a73344fe94a/ff44c/salty_egg.jpg 158w, /static/8058f3f26913fea3b6a89a73344fe94a/a6688/salty_egg.jpg 315w, /static/8058f3f26913fea3b6a89a73344fe94a/828fb/salty_egg.jpg 630w, /static/8058f3f26913fea3b6a89a73344fe94a/0ede0/salty_egg.jpg 945w, /static/8058f3f26913fea3b6a89a73344fe94a/3ac88/salty_egg.jpg 1260w, /static/8058f3f26913fea3b6a89a73344fe94a/e1596/salty_egg.jpg 2048w" sizes="(max-width: 630px) 100vw, 630px" loading="lazy" decoding="async" style="width: 100%; height: 100%; margin: 0px; vertical-align: middle; position: absolute; top: 0px; left: 0px; opacity: 1; transition: opacity 0.5s ease 0s; color: inherit; box-shadow: white 0px 0px 0px 400px inset;">`

### Actual Result

`<img class="gatsby-resp-image-image" alt="Chinese Salty Egg" title="" src="/static/8058f3f26913fea3b6a89a73344fe94a/828fb/salty_egg.jpg" srcset="/static/8058f3f26913fea3b6a89a73344fe94a/ff44c/salty_egg.jpg 158w /static/8058f3f26913fea3b6a89a73344fe94a/a6688/salty_egg.jpg 315w /static/8058f3f26913fea3b6a89a73344fe94a/828fb/salty_egg.jpg 630w /static/8058f3f26913fea3b6a89a73344fe94a/0ede0/salty_egg.jpg 945w /static/8058f3f26913fea3b6a89a73344fe94a/3ac88/salty_egg.jpg 1260w /static/8058f3f26913fea3b6a89a73344fe94a/e1596/salty_egg.jpg 2048w" sizes="(max-width: 630px) 100vw, 630px" loading="lazy" decoding="async" style="width: 100%; height: 100%; margin: 0px; vertical-align: middle; position: absolute; top: 0px; left: 0px; opacity: 1; transition: opacity 0.5s ease 0s; color: inherit; box-shadow: white 0px 0px 0px 400px inset;">`

### Environment

```shell
All default from `gatsby-starter-blog` but with `rehype-react` added:

  System:
    OS: macOS 12.1
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 96.0.4664.110
    Safari: 15.2
  npmPackages:
    gatsby: ^4.4.0 => 4.4.0
    gatsby-plugin-feed: ^4.4.0 => 4.4.0
    gatsby-plugin-gatsby-cloud: ^4.4.0 => 4.4.0
    gatsby-plugin-google-analytics: ^4.4.0 => 4.4.0
    gatsby-plugin-image: ^2.4.0 => 2.4.0
    gatsby-plugin-manifest: ^4.4.0 => 4.4.0
    gatsby-plugin-offline: ^5.4.0 => 5.4.0
    gatsby-plugin-react-helmet: ^5.4.0 => 5.4.0
    gatsby-plugin-sharp: ^4.4.0 => 4.4.0
    gatsby-remark-copy-linked-files: ^5.4.0 => 5.4.0
    gatsby-remark-images: ^6.4.0 => 6.4.0
    gatsby-remark-prismjs: ^6.4.0 => 6.4.0
    gatsby-remark-responsive-iframe: ^5.4.0 => 5.4.0
    gatsby-remark-smartypants: ^5.4.0 => 5.4.0
    gatsby-source-filesystem: ^4.4.0 => 4.4.0
    gatsby-transformer-remark: ^5.4.0 => 5.4.0
    gatsby-transformer-sharp: ^4.4.0 => 4.4.0
```

**Comment by Ameobea** [comment]
I'm currently using this function as a workaround to patch the `htmlAst` before handing it off to `rehype-react`:

```ts
const patchHTMLAST = (ast) => {
  // There is a funny bug somewhere in one of the dozens-hundreds of libraries being used to transform markdown which
  // is causing a problem with `srcset`.  Images are being converted into HTML in the markdown which automatically
  // references the sources of the generated resized/converted images.  The generated `srcset` prop is being somehow
  // converted into an array of strings before being passed to `rehype-react` which is then concatenating them and
  // generating invalid HTML.
  //
  // This code handles converting `srcset` arrays into valid strings.
  if (ast.properties?.srcSet && Array.isArray(ast.properties.srcSet)) {
    ast.properties.srcSet = ast.properties.srcSet.join(', ');
  }
  if (ast.children) {
    ast.children = ast.children.map(patchHTMLAST);
  }
  return ast;
};
```

**Comment by Ameobea** [comment]
This issue is still active.  I've not directly verified it again since posting it, but I have no reason to expect things would have changed.

**Comment by yathomasi** [comment]
Linking here for further debugging the issue: https://github.com/rehypejs/rehype-react/issues/41#issuecomment-1197818406

### Reproducible link 
[codesandbox:gatsby-starter-blog-rehype-react](https://codesandbox.io/s/gatsby-starter-blog-rehype-react-95q31j?file=/src/templates/blog-post.js)

### Steps to reproduce
Adding react-rehype to parse the htmlAst in the existing [gatsbyjs/gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog)

**Comment by essenmitsosse** [comment]
I can confirm this issue is still there and is quite a hard break for the plugin — in a way that might go unnoticed by many people (which isn't a good thing)

**Comment by danilobuerger** [comment]
So I hunted this down:

The issue is caused by the package `property-information` declaring `srcSet` as comma separated:

https://github.com/wooorm/property-information/blob/19e59d5a60723b57ad07c51982bc950652537f6b/lib/html.js#L226

This was fixed in the latest release (`6.1.1`) of `property-information`:

https://github.com/wooorm/property-information/commit/acf30dd81942e1c03e5ea4301887307df4c6ba30

The problem now is that `gatsby-transformer-remark` depends on an old version (`^6.0.2`) of `hast-util-raw`. Through the dependency chain, an old version of `property-information` is included:
```
=> Found "property-information@5.6.0"
info Has been hoisted to "property-information"
info Reasons this module exists
   - Hoisted from "gatsby-transformer-remark#hast-util-to-html#property-information"
   - Hoisted from "gatsby-transformer-remark#hast-util-raw#hast-util-from-parse5#property-information"
   - Hoisted from "gatsby-transformer-remark#hast-util-raw#hast-util-to-parse5#property-information"
   - Hoisted from "gatsby-transformer-remark#hast-util-raw#hast-util-from-parse5#hastscript#property-information"
   - Hoisted from "gatsby-transformer-remark#hast-util-raw#hast-util-to-parse5#hast-to-hyperscript#property-information"
```

# Short term Fix

Apply this patch with `patch-package`:
```patch
diff --git a/node_modules/property-information/lib/html.js b/node_modules/property-information/lib/html.js
index 1db1ef9..b2df2ea 100644
--- a/node_modules/property-information/lib/html.js
+++ b/node_modules/property-information/lib/html.js
@@ -80,7 +80,7 @@ module.exports = create({
     httpEquiv: spaceSeparated,
     id: null,
     imageSizes: null,
-    imageSrcSet: commaSeparated,
+    imageSrcSet: null,
     inputMode: null,
     integrity: null,
     is: null,
@@ -223,7 +223,7 @@ module.exports = create({
     src: null,
     srcDoc: null,
     srcLang: null,
-    srcSet: commaSeparated,
+    srcSet: null,
     start: number,
     step: null,
     style: null,
```

# Long term fix

Someone should take the time and update the dependencies in `gatsby-transformer-remark`.
